circle.yml

```diff
@@ -0,0 +1,43 @@
+machine:
+  timezone:
+    GMT
+  xcode:
+    version: 8.0
+  environment:
+    XCODE_SCHEME: nupic
+    XCODE_WORKSPACE: nupic
+    ARCHFLAGS: "-arch x86_64"
+    PYTHONPATH: "$HOME/Library/Python/2.7/lib/python/site-packages"
+    PATH: "$HOME/Library/Python/2.7/bin:$PATH"
+    PYBIN: "$HOME/Library/Python/2.7/bin"
+  pre:
+    - pip install --user --ignore-installed --upgrade --verbose virtualenv
+    - ln -s ~/Library/Python/2.7/bin/virtualenv /usr/local/bin/virtualenv
+
+dependencies:
+  override:
+    - curl --silent --show-error --retry 5 -O http://releases.numenta.org/pip/1ebd3cb7a5a3073058d0c9552ab074bd/get-pip.py
+    - python get-pip.py --user --upgrade --ignore-installed pip setuptools wheel
+    # Still don't get the user-installed version of pip, but this seems manageable.
+    - pip --version
+    - pip install --use-wheel --user -r bindings/py/requirements.txt --quiet || exit
+    - pip install --user pycapnp==0.5.8
+    - pip install --user pytest
+
+compile:
+  override:
+    - mkdir -p build/scripts
+    - cd build/scripts && cmake ../.. -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=../release -DPY_EXTENSIONS_DIR=../../bindings/py/nupic/bindings
```

Comment:
Similarly, consider moving into a standalone script invoked here.  That would make it easy to do something like the following, which I find cleaner:

```
pushd build/scripts
cmake ...
make -j6
make install
popd
```